.github/workflows/api_changes.yml

```diff
@@ -0,0 +1,102 @@
+# This CI configuration uses griffe to detect api changes. When a PR is opened,
+# this will check the PR branch against its HEAD. If changes are detected, they 
+# will be displayed as a PR comment and uploaded as an artifact. A label of 
+# `api_changes` will also be added to the PR.
+# This workflow also includes a workflow_run to manually trigger checks against 
+# arbitrary tags or commits.
+name: API changes
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      base_ref:
+        description: "Baseline commit or git reference"
+        required: true
+      contender_ref:
+        description: "Contender commit or git reference"
+        required: true
+
+jobs:
+  griffe:
+    runs-on: ubuntu-latest
+    steps:
+      # We need the full repo to avoid this issue
+      # https://github.com/actions/checkout/issues/23
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: "3.9"
```

Comment:
Any particular reason why we use 3.9 vs (any other supported version) 

When should we update this :) ?

Reply:
I just added it to ensure consistency between runners (as per docs). I can add a comment. I guess it can be updated to 3.10 for that matter.